### PR TITLE
Hotfix/1.1 misc

### PR DESCRIPTION
**Pull request by maddyscientist**
This PR is a collection of fixes, some of which are back ported from the GK branch.  This PR has been tested against BQCD, TIFR SU3 and MILC HISQ RHMC (both QMP and MPI).
* Fix staggered dslash kernel to correctly issue FMA instructions
* Reduce peak memory consumption of `computeKSLinkQuda`
* Remove unused memory allocation from `staggeredForceQuda`
* Check thread count is valid for multi-GPU gauge fixing OVR
* Add extended radius to `GaugeField::aux_string` to break degeneracy
* Fix false positive valgrind warnings
* Fix memory initialization issue with deflation
* Fix unitarize_link_test which broke with split grid (`endQuda` called prematurely)
* Improve robustness of link comparisons to check for nan
* Fix bugs in the Fortran interface
* Add `struct_size` parameter to interface structs to check for consistency between internal and external struct definitions (helps catch breakages in the Fortran interface)
* Fix UB in the GaugeField checksum kernel (fixes BQCD for CUDA 11.x)
* Correctly use base-2 units for printing QUDA memory statistics in `endQuda` 
* Update QMP to 2.5.3
* Increase max number of MG levels to 5

### CODE REVIEW

**Review by hummingtree — APPROVED** [review]

**Review by mathiaswagner — APPROVED** [review]

File: include/malloc_quda.h [hunk 1]
```diff
@@ -9,25 +9,45 @@ namespace quda {
...
+
+  /**
+     @return mapped memory allocated
+   */
+  size_t mapped_allocated();
+
+  /**
+     @return host memory allocated
+   */
+  size_t host_allocatedk();
```
Comment (weinbe2) [inline include/malloc_quda.h]: Is the `k` at the end of this a typo?

File: include/color_spinor.h [hunk 2]
```diff
@@ -1192,6 +1192,42 @@ namespace quda {
     return y;
   }
 
+  /**
+     @brief Compute the matrix-vector product z = A * x + y
+     @param[in] A Input matrix
+     @param[in] x Input vector
+     @param[in] y Input vector
+     @return The vector A * x
```
Comment (weinbe2) [inline include/color_spinor.h]: `A * x + y`, not just `A * x`

File: include/color_spinor.h [hunk 3]
```diff
@@ -1192,6 +1192,42 @@ namespace quda {
...
+  */
+  template<typename Float, int Nc, int Ns> __device__ __host__ inline
+  ColorSpinor<Float,Nc,Ns> mv_add(const Matrix<complex<Float>,Nc> &A, const ColorSpinor<Float,Nc,Ns> &x, const ColorSpinor<Float,Nc,Ns> &y)
```
Comment (weinbe2) [inline include/color_spinor.h:1203]: Should we have a version that just does `A * x` (`mv_add(...A, ...x);`), just to have it handy (and maybe as a micro-optimization somewhere)?

Comment (weinbe2) [reply include/color_spinor.h:1203]: Addressed offline, it's indeed what `operator*` does

**Review by weinbe2 — APPROVED** [review]
All looks great --- this is good to merge.

**Pull request merged**